src/Lean/Meta/SynthInstance.lean

```diff
@@ -102,7 +102,7 @@ partial def normLevel (u : Level) : M Level := do
     | Level.max v w     => return u.updateMax! (← normLevel v) (← normLevel w)
     | Level.imax v w    => return u.updateIMax! (← normLevel v) (← normLevel w)
     | Level.mvar mvarId =>
-      if !(← isLevelMVarAssignable mvarId) then
+      if (← getMCtx).getLevelDepth mvarId != (← getMCtx).depth then
```

Comment:
This was an oversight in #1953.  The `isLevelMVarAssignable` used to check for depth equality, irrespective of whether `ignoreLevelMVarDepth` was set or not.  (While other functions like `isReadOnly` took it into account.)  Now these functions all use the same logic.